algo.lua

```diff
@@ -0,0 +1,137 @@
+local ffi = require 'ffi'
+local errcheck = cudnn.errcheck
+
+local algo = {}
+local autotunerCache = {}
+autotunerCache[1] = {} -- forward
+autotunerCache[2] = {} -- backwardFilter
+autotunerCache[3] = {} -- backwardData
+
+local function setupAlgo(self, algo_t, perf_t, findAPI, getAPI, wsAPI, algSearchMode, params)
+
+        local algType = ffi.new(algo_t, 1)
+
+        if cudnn.benchmark or cudnn.fastest then -- the manual auto-tuner is run
+            if autotunerCache[1][self.autotunerHash] then
```

Comment:
You are absolutely right, I am surprised it even worked! I will use findAPI as the key instead of numbers.